python/uv/__init__.py

```diff
@@ -28,3 +28,21 @@
     ]
 else:
     __all__ = ["find_uv_bin"]
+
+
+def __getattr__(attr_name: str) -> object:
+    if attr_name in {
+        "build_sdist",
+        "build_wheel",
+        "build_editable",
+        "get_requires_for_build_sdist",
+        "get_requires_for_build_wheel",
+        "prepare_metadata_for_build_wheel",
+        "get_requires_for_build_editable",
+        "prepare_metadata_for_build_editable",
+    }:
+        err = f"Using `uv.{attr_name}` is not allowed. The uv build backend requires preview mode to be enabled, e.g., via the `UV_PREVIEW=1` environment variable."
```

Comment:
If you're hitting this branch, you need `UV_PREVIEW=1`, that's e.g. with `python -m build`. For `uv build`, you usually want `--preview` which goes through the direct path, except for `uv build --force-pep517`, in which case you need `UV_PREVIEW=1`, which is why I had just put "Use `UV_PREVIEW=1` and `--preview`" on top of the docs.

We could also set preview through PEP 517 config settings, but that seems even worse.

Reply:
Just to confirm, is this asking for a change to what I wrote here?

Reply:
I was only adding context on why this preview feature is complicated beyond the usual "Add `--preview` to use".

Reply:
👍 sounds good, that makes sense to me.